zerver/management/commands/deliver_scheduled_messages.py

```diff
@@ -44,7 +48,29 @@ def handle(self, *args: Any, **options: Any) -> None:
                             scheduled_message.scheduled_timestamp,
                             scheduled_message.sender_id,
                         )
-                        send_scheduled_message(scheduled_message)
+                        try:
+                            send_scheduled_message(scheduled_message)
+                        except JsonableError as e:
+                            scheduled_message.failed = True
+                            scheduled_message.failure_message = e.msg
+                            scheduled_message.save(update_fields=["failed", "failure_message"])
+                            logging.info(
+                                "Failed with message: %s", scheduled_message.failure_message
+                            )
+                        except Exception:
+                            # An unexpected failure; store as a generic 500 error.
+                            scheduled_message.refresh_from_db()
+                            was_delivered = scheduled_message.delivered
+                            scheduled_message.failed = True
+                            scheduled_message.failure_message = _("Internal server error")
```

Comment:
I don't think this knows the user's language to localize correctly.

Reply:
(a contextmanager off the user's preferences somehow can certainly make this work, but I don't think it works as written)

Reply:
The JsonableError branh above probably has the same problem, but the localization is hidden inside the raise. inside `send_scheduled_message`, so any contextmanager fix should be above the `try`

Reply:
I think you're right but the function of `failed_message` might very well be just for our own debugging anyway; let me discuss in a product/design thread, but I think we might want to just use the `failed` boolean with clients.

Reply:
If it's for internal debugging, then we shouldn't be throwing away information and using a generic phrase here.

Reply:
https://github.com/zulip/zulip/issues/25501 tracks the work to notify users in a reasonable way, which will surely involve refactoring this.